src/coreclr/dlls/mscorpe/pewriter.cpp

```diff
@@ -1926,7 +1932,11 @@ HRESULT PEWriter::fixup(CeeGenTokenMapper *pMapper)
                     is.NumberOfAuxSymbols = 1;
                     if((pch = reloc->getBlock(sizeof(IMAGE_SYMBOL))))
                         memcpy(pch,&is,sizeof(IMAGE_SYMBOL));
-                    else return E_OUTOFMEMORY;
+                    else
+                    {
+                        delete[] TokInSymbolTable;
+                        return E_OUTOFMEMORY;
```

Comment:
We have also allocated `reloc` and `strtab`. Should we delete and set them to null too?

Reply:
Is this method and other stuff related to fixups in this directory reachable code?

It all looks related to `EMIT_FIXUPS` that is never defined. It would be better to delete it all.

Reply:
> Is this method and other stuff related to fixups in this directory reachable code?

This memory leak was flagged by a static analyzer. I am not sure it can tell if the code is reachable or not.

Reply:
Could you please try to delete this method, all code under `EMIT_FIXUPS` ifdefs, and see whether it will compile fine (maybe with a few other unrechable pieces deleted)?

Reply:
I tried removing `#ifdef EMIT_FIXUPS` (and the code they guard) completely, but this `CeeFileGenWriter::fixup` fails to build looking for this (now undefined) method:

https://github.com/dotnet/runtime/blob/400311b032fe5d05b49fb6a813e4a5a60604d8dd/src/coreclr/dlls/mscorpe/ceefilegenwriter.cpp#L518

That call is not guarded by an `#ifdef EMIT_FIXUPS`...

Reply:
Ok, it was a bit less straightforward than I hoped. I have submitted #50259

Reply:
Thanks! I am going to close this issue in favour of #50259 then

Reply:
I think a few of the places you have touched will be still around even after my PR is merged. We can keep this one open, and then you can just resolve the conflict.